zk_prover/src/merkle_sum_tree/mst.rs

```diff
@@ -133,6 +113,14 @@ impl<const N_CURRENCIES: usize, const N_BYTES: usize> MerkleSumTree<N_CURRENCIES
 
         let mut nodes = vec![];
 
+        // Pad the entries with empty entries to make the number of entries equal to 2^depth
+        if entries.len() < 2usize.pow(depth as u32) {
```

Comment:
I wonder if instead of stuffing the rest of the tree we could only add the zero node when the right hasher input is missing on any level. This way there would be less computation over the empty branches.

Reply:
Thanks for the idea, opened an issue 4 that!